package-parser/package_parser/processing/api/model/_api.py

```diff
@@ -292,14 +302,49 @@ def to_json(self) -> Any:
             ],
         }
 
-    def get_formatted_code(self) -> str:
+    def get_formatted_code(self, *, cut_documentation: bool = False) -> str:
         if self.formatted_code is None:
             self.formatted_code = _generate_formatted_code(self)
+            if cut_documentation:
+                self.formatted_code = _cut_documentation_from_code(
+                    self.formatted_code, self
+                )
         return self.formatted_code
 
     def __repr__(self) -> str:
         return "Class(id=" + self.id + ")"
 
+    def __hash__(self) -> int:
+        return hash(
+            (
+                self.id,
+                self.qname,
+                frozenset(self.decorators),
+                frozenset(self.superclasses),
+                frozenset(self.methods),
+                self.is_public,
+                frozenset(self.reexported_by),
+                self.documentation,
+                self.code,
+                frozenset(self.instance_attributes),
+            )
+        )
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, Class)
+            and self.id == other.id
+            and self.qname == other.qname
+            and self.decorators == other.decorators
+            and self.superclasses == other.superclasses
+            and self.methods == other.methods
+            and self.is_public == other.is_public
+            and self.reexported_by == other.reexported_by
+            and self.documentation == other.documentation
+            and self.code == other.code
+            and self.instance_attributes == other.instance_attributes
+        )
+
```

Comment:
`__hash__` and `__eq__` already get created for a dataclass. Is there a particular reason you have to override them?

Reply:
For parameters: The necessary frozen attribute that is needed for the creation of a hash function would conflict with the creation of types in the init function if frozen is set True.
For all other cases, `__eq__` and `__hash__` will be removed and frozen will be set True for Safe-DS/Library-Analyzer#25